server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1402,6 +1402,20 @@ private void closeConsistencyCheck() {
         throw new RuntimeException(msg);
       }
 
+      if (tabletMeta.getFlushId().orElse(-1) != lastFlushID) {
```

Comment:
Last week I examined to lastFlushID in the tablet server code to see if it is -1 when not set.  Based on my analysis, it is -1 so doing orElse(-1) is ok here.  I don't think I analyzed lastCompactID.  If it has not been done, it would be good to analyze initialization of both of them as my analysis was quick. 

This -1 practice is extremely brittle.  A good follow on PR would be to do something else (constants, OptionalLong, ??) in a follow on PR to make it less brittle.

Reply:
I will go ahead and analyze the initialization of both lastFlushID and lastCompactID.

Reply:
Is there more to be done here? I see @keith-turner approved this. Is this ready to merge?